Use of `std::make_unique` with `-std=c++11`

Currently, CMakeLists.txt says the minimum supported C++ standard is C++ 11, but examples/accel/OpenCLcontext.cpp uses `std::make_unique` which is only available with C++ 14 and newer.

Originally hit this downstream in Gentoo at https://bugs.gentoo.org/903968:
```
[1932/1951] /var/tmp/portage/dev-libs/pocl-3.1/temp/clang-wrappers/x86_64-pc-linux-gnu-g++ -DCL_HPP_TARGET_OPENCL_VERSION=300 -DCL_TARGET_OPENCL_VERSION=300 -DCL_USE_DEPRECATED_OPENCL_1_0_APIS -DCL_USE_DEPRECATED_OPENCL_1_1_APIS -DCL_USE_DEPRECATED_OPENCL_1_2_APIS -DCL_USE_DEPRECATED_OPENCL_2_0_APIS -DCL_USE_DEPRECATED_OPENCL_2_1_APIS -DCL_USE_DEPRECATED_OPENCL_2_2_APIS -DSRCDIR=\"/var/tmp/portage/dev-libs/pocl-3.1/work/pocl-3.1/examples/accel\" -I/var/tmp/portage/dev-libs/pocl-3.1/work/pocl-3.1_build -I/var/tmp/portage/dev-libs/pocl-3.1/work/pocl-3.1/include -I/var/tmp/portage/dev-libs/pocl-3.1/work/pocl-3.1/include/hpp -I/var/tmp/portage/dev-libs/pocl-3.1/work/pocl-3.1/poclu -I/var/tmp/portage/dev-libs/pocl-3.1/work/pocl-3.1/lib/CL -I/var/tmp/portage/dev-libs/pocl-3.1/work/pocl-3.1/examples/accel  -O2 -pipe -march=native -fdiagnostics-color=always -frecord-gcc-switches -Wreturn-type  -ggdb3 -std=gnu++14 -fPIE -fdiagnostics-color=always -Wno-ignored-attributes -fvisibility=hidden -fvisibility-inlines-hidden -std=c++11 -Wno-deprecated -Wno-deprecated-declarations -MD -MT examples/accel/CMakeFiles/accel_example.dir/accel_example.cpp.o -MF examples/accel/CMakeFiles/accel_example.dir/accel_example.cpp.o.d -o examples/accel/CMakeFil
/var/tmp/portage/dev-libs/pocl-3.1/work/pocl-3.1/examples/accel/OpenCLcontext.h:30: warning: "CL_HPP_TARGET_OPENCL_VERSION" redefined
   30 | #define CL_HPP_TARGET_OPENCL_VERSION 120
      |
<command-line>: note: this is the location of the previous definition
/var/tmp/portage/dev-libs/pocl-3.1/work/pocl-3.1/include/hpp/CL/cl2.hpp:18:151: note: ‘#pragma message: cl2.hpp has been renamed to opencl.hpp to make it clear that it supports all versions of OpenCL. Please include opencl.hpp directly.’
   18 | #pragma message("cl2.hpp has been renamed to opencl.hpp to make it clear that it supports all versions of OpenCL. Please include opencl.hpp directly.")
      |                                                                                                                                                       ^
/var/tmp/portage/dev-libs/pocl-3.1/work/pocl-3.1/examples/accel/OpenCLcontext.cpp: In constructor ‘OpenCL_Manager::OpenCL_Manager()’:
/var/tmp/portage/dev-libs/pocl-3.1/work/pocl-3.1/examples/accel/OpenCLcontext.cpp:192:49: error: ‘make_unique’ is not a member of ‘std’
  192 | OpenCL_Manager::OpenCL_Manager() : Context{std::make_unique<OpenCL_Context>()} {}
      |                                                 ^~~~~~~~~~~
/var/tmp/portage/dev-libs/pocl-3.1/work/pocl-3.1/examples/accel/OpenCLcontext.cpp:192:49: note: ‘std::make_unique’ is only available from C++14 onwards
/var/tmp/portage/dev-libs/pocl-3.1/work/pocl-3.1/examples/accel/OpenCLcontext.cpp:192:75: error: expected primary-expression before ‘>’ token
  192 | OpenCL_Manager::OpenCL_Manager() : Context{std::make_unique<OpenCL_Context>()} {}
      |                                                                           ^
/var/tmp/portage/dev-libs/pocl-3.1/work/pocl-3.1/examples/accel/OpenCLcontext.cpp:192:77: error: expected primary-expression before ‘)’ token
  192 | OpenCL_Manager::OpenCL_Manager() : Context{std::make_unique<OpenCL_Context>()} {}
      |                                                                             ^
/var/tmp/portage/dev-libs/pocl-3.1/work/pocl-3.1/examples/accel/OpenCLcontext.cpp:192:36: error: no matching function for call to ‘std::unique_ptr<OpenCL_Context>::unique_ptr(<brace-enclosed initializer list>)’
  192 | OpenCL_Manager::OpenCL_Manager() : Context{std::make_unique<OpenCL_Context>()} {}
      |                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/examples/accel/OpenCLcontext.cpp
+++ b/examples/accel/OpenCLcontext.cpp
@@ -189,7 +189,8 @@ bool OpenCL_Manager::processCameraFrame(unsigned char* input, unsigned long *out
     return isValid;
 }
 
-OpenCL_Manager::OpenCL_Manager() : Context{std::make_unique<OpenCL_Context>()} {}
+OpenCL_Manager::OpenCL_Manager()
+    : Context{std::unique_ptr<OpenCL_Context>(new OpenCL_Context())} {}
 OpenCL_Manager::~OpenCL_Manager() {}
 
 

